core/lib/snt/core/services/result.rb

```diff
@@ -4,13 +4,19 @@ module Services
       class Result
         attr_accessor :data, :errors, :warnings, :events
 
+        WARNING = 'WARNING'.freeze
+
         def initialize
           @data = {}
           @errors = []
           @warnings = []
           @events = []
         end
 
+        def add_warning(message, code = WARNING)
```

Comment:
We should try to make the warnings and errors have the same method signature.  It will be confusing that one is message, code and the other is code, message.